rust-stream/src/main.rs

```diff
@@ -0,0 +1,413 @@
+use std::fmt::{Debug, Display};
```

Comment:
Consider adding an inner doc comment (//!) here to describe the goals and objectives of this CLI tool.

As an outsider, it helps to get an overview.